tools/clang/test/HLSLFileCheck/passes/hl/mem2reg_hlsl/mem2reg_simple_split.ll

```diff
@@ -0,0 +1,182 @@
+; RUN: %opt %s -dxil-cond-mem2reg -S | FileCheck %s
+
+; CHECK: @main
+; CHECK-NOT: alloca [2 x float]
+
+; CHECK-DAG: !DIExpression(DW_OP_bit_piece, 0, 32)
+; CHECK-DAG: !DIExpression(DW_OP_bit_piece, 32, 32)
```

Comment:
Maybe checks to verify that the GEPs are gone and replaced with the new allocas? I know that might get complicated, but it would more fully vet the behavior of this pass, making it a more complete unit test.

Reply:
Thanks! Modified this test to make sure there's no allocas at all. The point of split array that early was to get get rid of the allocas.